python/lsst/ts/externalscripts/maintel/park_calibration_projector.py

```diff
@@ -0,0 +1,131 @@
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+__all__ = ["ParkCalibrationProjector"]
+
+import yaml
+from lsst.ts import salobj
+from lsst.ts.observatory.control.maintel.mtcalsys import MTCalsys
+
+
+class ParkCalibrationProjector(salobj.BaseScript):
+    """Move the calibration projector into a safe position
+
+    Parameters
+    ----------
+    index : int
+        Index of Script SAL component.
+    """
+
+    def __init__(self, index):
+        super().__init__(
+            index=index,
+            descr="Park Calibration Projector",
+        )
+
+        self.mtcalsys = None
+
+    def set_metadata(self, metadata):
+        metadata.duration = 30
+
+    @classmethod
+    def get_schema(cls):
+        schema_yaml = """
+            $schema: http://json-schema.org/draft-07/schema#
+            $id: https://github.com/lsst-ts/ts_externalscripts/maintel/calibrations/park_calibration_projector.yaml # noqa: E501
+            title: ParkCalibrationProjector v1
+            description: Park the Calibration after use to ensure LEDs are off and stages
+                are in a safe plce
+            type: object
+            properties:
+                ignore:
+                    description: >-
+                        CSCs from teh group to ignore in status check
+                    type: array
+                    items:
+                        type: string
+
+            additionalProperties: false
+        """
+        return yaml.safe_load(schema_yaml)
+
+    async def configure(self, config):
+        """Configure the script.
+
+        Parameters
+        ----------
+        config : ``self.cmd_configure.DataType``
+
+        """
+        self.log.info("Configure started")
+        if self.mtcalsys is None:
+            self.log.debug("Creating MTCalSys.")
+            self.mtcalsys = MTCalsys(domain=self.domain, log=self.log)
+            await self.mtcalsys.start_task
+
+        if hasattr(config, "ignore"):
+            self.mtcalsys.disable_checks_for_components(components=config.ignore)
+
+        self.log.info("Configure completed")
+
+    async def run(self):
+        """Run script."""
+        await self.assert_components_enabled()
+
+        self.log.info("Parking Calibration Projector")
+        await self.mtcalsys.park_projector()
+
+        await self.mtcalsys.get_projector_setup()
+
+        led_location = await self.mtcalsys.linearstage_led_select.tel_position.next(
+            flush=True
+        )
+        self.log.debug(f"LED Location: {led_location}")
+        assert led_location == self.mtcalsys.led_rest_position
+
+    async def assert_components_enabled(self):
```

Comment:
No need for a specific method here, you can just call self.mtcalsys.assert_all_enabled() directly.